lib/poller.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+class Poller {
+  constructor ({
+    pollingRateSeconds,
+    serializeSegments,
+    splitioApi
+  }) {
+    this.cache = {}
+    this.pollingRateSeconds = pollingRateSeconds || 300
+    this.serializeSegments = serializeSegments || false
+    this.splitioApi = splitioApi
+  }
+
+  async pollForChanges () {
+    try {
+      const splitChanges = await this.splitioApi.getSplitChanges()
+      Object.assign(this.cache, splitChanges)
+    } catch (err) {
+      console.log('Error getting split changes', err)
```

Comment:
this is why an event emitter is useful.  you can emit an `error` event and the user of this class can decide how they want to handle errors (e.g., ignoring them, or logging them using whatever logger they have).